lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -474,6 +469,94 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		return $user_cpt;
 	}
 
+	// @codingStandardsIgnoreStart Squiz.Commenting.FunctionComment.ParamCommentFullStop
+	/**
+	 * Saves a new user variation into the database.
+	 *
+	 *
+	 * @param array $args Arguments. All are required.
+	 *     {
+	 *     @type string title Global styles variation name.
+	 *     @type string global_styles Global styles settings as a JSON string.
+	 *     @type string $stylesheet Slug of the theme associated with these global styles.
+	 * }
+	 *
+	 * @return int|WP_Error Post ID of the new variation or error if insertion failed.
+	 */
+	public static function add_user_global_styles_variation( $args ) {
+		$theme = wp_get_theme();
+
+		/*
+		 * Bail early if the theme does not support a theme.json.
+		 *
+		 * Since wp_theme_has_theme_json only supports the active
+		 * theme, the extra condition for whether $theme is the active theme is
+		 * present here.
+		 */
+		if ( $theme->get_stylesheet() === $args['stylesheet'] && ! wp_theme_has_theme_json() ) {
+			return new WP_Error( __( 'Theme does not have theme.json', 'gutenberg' ) );
+		}
+
+		$post_id = wp_insert_post(
+			array(
+				'post_content' => $args['global_styles'],
+				'post_status'  => 'publish',
+				'post_title'   => $args['title'],
+				'post_type'    => 'wp_global_styles',
+				'post_name'    => sprintf( 'wp-global-styles-%s', urlencode( $args['stylesheet'] ) ),
```

Comment:
Can posts of the same type (`wp_global_styles`) share the same `post_name`? Thinking we could use something along the lines of `wp-global-styles-stylesheet-variation-urlencode(title)`. I presume two variations of the same theme shouldn't have the same title. We should check for that (also in the client).